connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/external/ExternalConnectorExecutor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.core.external;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.api.secret.SecretProvider;
+import io.camunda.connector.api.validation.ValidationProvider;
+import io.camunda.connector.runtime.core.outbound.OutboundConnectorFactory;
+
+public class ExternalConnectorExecutor {
+
+  private final OutboundConnectorFactory outboundConnectorFactory;
+
+  private final SecretProvider secretProvider;
+  private final ValidationProvider validationProvider;
+  private final ObjectMapper objectMapper;
+
+  public ExternalConnectorExecutor(
+      OutboundConnectorFactory outboundConnectorFactory,
+      SecretProvider secretProvider,
+      ValidationProvider validationProvider,
+      ObjectMapper objectMapper) {
+    this.outboundConnectorFactory = outboundConnectorFactory;
+    this.secretProvider = secretProvider;
+    this.validationProvider = validationProvider;
+    this.objectMapper = objectMapper;
+  }
+
+  public Object execute(String type, Object variables) throws Exception {
+    var connector = outboundConnectorFactory.getInstance(type);
+    String variablesJson;
+    try {
+      variablesJson = objectMapper.writeValueAsString(variables);
+    } catch (Exception e) {
+      throw new RuntimeException("Error while serializing variables", e);
+    }
```

Comment:
If we transform the `variables` to json wont we have an issue with passing data around like a byte[] for example?

Reply:
Do you have a specific scenario in mind? As I'm thinking about it, it seems that we will be fine if we:
- only pass binary data as part of the `Document` class, and
- have a proper serialization/deserialization mechanism in place for `Document` that converts it to base64.

I might be missing something, and it is possible that we can do secret replacement more efficiently 🤔

Reply:
@chillleader Sure, lets assume we want to execute an S3 connector in the `resultExpression` of the REST Connector to store the binary body. Passing a byte[] document wouldnt really be efficiently possible if we serialize it to json first and read it back again for larger files. Here it would be better to be able to directly invoke the connector passing the objects/references in memory to be able to use streams for example.

Yes, secret replacing would be an issue but this could theoretically be done by only replacing it when an object in the variables is of type `string` before passing it to the Connector.

Reply:
Thanks for clarifying, makes sense. This requires refactoring how we replace secrets in all outbound connectors to avoid having two secret replacement methods (we are still getting a json string from the zeebe job). I'm not opposed to that and can also attempt this as part of this PR.

Reply:
Great, you can also add secret support later on to figure out whether our approach here is reasonable 😄

Reply:
Changed it here :)